internal/pkg/services/iaas/utils/utils.go

```diff
@@ -98,3 +99,12 @@ func GetNetworkRangeFromAPIResponse(prefix string, networkRanges *[]iaas.Network
 	}
 	return iaas.NetworkRange{}, fmt.Errorf("new network range not found in API response")
 }
+
+// GetGroupName retrieves the name of a security group for a project
+func GetGroupName(ctx context.Context, apiClient IaaSClient, projectId string, securityGroupId string) (string, error) {
```

Comment:
`GetSecurityGroupName` would be better as there might be other types of groups

Reply:
Renamed the function and the testcase